command/ca/provisionerbeta/provisioner.go

```diff
@@ -238,4 +241,44 @@ By default it will accept any SAN in the CSR.`,
 with the same instance will be accepted. By default only the first request
 will be accepted.`,
 	}
+
+	// Nebula provisioner flags
+	nebulaRootFlag = cli.StringFlag{
+		Name: "nebula-root",
+		Usage: `Root certificate (chain) <file> used to validate the signature on Nebula
+provisioning tokens.`,
+	}
 )
+
+func readNebulaRoots(rootFile string) ([][]byte, error) {
+	b, err := utils.ReadFile(rootFile)
+	if err != nil {
+		return nil, err
+	}
+
+	var crt *nebula.NebulaCertificate
+	var certs []*nebula.NebulaCertificate
+	for len(b) > 0 {
+		crt, b, err = nebula.UnmarshalNebulaCertificateFromPEM(b)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error reading %s", rootFile)
+		}
+		if crt.Details.IsCA {
+			certs = append(certs, crt)
+		}
+	}
+	if len(certs) == 0 {
+		return nil, errors.Errorf("error reading %s: no certificates found", rootFile)
```

Comment:
This error case is when no CA certificates were found, right?

Reply:
Yes, I'll use this:
```golang
return nil, errors.Errorf("error reading %s: no CA certificates found", rootFile)
```

Reply:
Fixed with c09e5d0